repo_test.go

```diff
@@ -27,14 +27,11 @@ func TestDuplicatedLinks(t *testing.T) {
 	query.Find("body  a").Each(func(_ int, s *goquery.Selection) {
 		href, ok := s.Attr("href")
 		if !ok {
-			log.Printf("expected '%s' href", s)
-			t.Fail()
+			t.Errorf("expected '%s' to have href", s.Text())
 		}
 
 		if links[href] {
-			log.Printf("duplicated link '%s'", href)
-			t.Fail()
-			return
+			t.Fatalf("duplicated link '%s'", href)
```

Comment:
Regarding this exit I am not quite sure if it would terminate the entire test (as its running inside a `Each` function but probably still the same gorutine.